### PR TITLE
Roloc updates

### PR DESCRIPTION
**Pull request by dmacattack**

### CODE REVIEW

File: rolocController.cpp [hunk 1]
```diff
@@ -296,18 +309,21 @@ void ROLOCcontroller::rolocSetParameters(quint16 mode, quint8 frequency)
...
 
-    /*qint8 status =*/  m_i2cBus.i2c_writeWord(mI2cAddr, LINEFINDER_VOLUME, data);
-    //todo:: consider error case for I2C
+    int status = m_i2cBus.i2c_writeWord(mI2cAddr, ROLOC::eCMD_VOLUME, data);
+    if (status != 0)
+    {
+        qWarning() << "i2c error (" << status << ")" << strerror(errno);
+    }
 
     mCurrVolume = data;
```
Comment (rabidmonkeyman) [inline rolocController.cpp:328]: if theres an error should the curr volume member variable be set?

File: rolocarrows.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * @brief ROLOCArrows::set - set the arrows from the i2c status byte
+ */
+void ROLOCArrows::set(quint8 statusByte)
+{
+    // reset the data
+    reset();
+
+    // acquire new data
+    mStatusByte = statusByte;
+    mbLeftArrow   = (statusByte >> LINEFINDER_LEFT_ARROW_BIT)   & 1;
```
Comment (rabidmonkeyman) [inline rolocarrows.cpp:33]: i assume this is all cut/paste from the previous code? theres gotta be a better, more readable way to do this

File: i2c.hpp [hunk 3]
```diff
@@ -37,7 +38,7 @@ class i2c
     int  i2c_writeWord(char hw_address, char address, int16_t data);
 
 private:
-
+    QMutex mI2cMutex;
```
[the remark this answers is not in the document]
Comment (dmacattack) [reply i2c.hpp:41]: duly noted

File: inspRolocControllerDbus.cpp [hunk 4]
```diff
@@ -64,14 +65,18 @@ void InspROLOCControllerDbus::sendDataReport(
...
     qDebug("DBus sending report");
+#endif
     emit rolocUpdateDataReport(mode, frequency, siglvl, depth, arrow, isPresent);
 }
 
 // sends data preset signal dbus -> UI
 void InspROLOCControllerDbus::sendPresent(bool isPresent)
 {
+#if DBG_BLOCK
     qDebug("DBus sending report");
```
[the remark this answers is not in the document]
Comment (dmacattack) [reply inspRolocControllerDbus.cpp:78]: yeah. good idea

File: rolocController.cpp (hunk 1 above)
Comment (dmacattack) [reply rolocController.cpp:328]: it should not. not i just stabbed that in there before i knew what a pile i was working with

File: rolocarrows.cpp (hunk 2 above)
Comment (dmacattack) [reply rolocarrows.cpp:33]: yeah copy paste. I did clean it up a bit. it was worse. I could change it to:
mbLeftArrow = ((statusByte & LINEFINDER_LEFT_ARROW_BIT) == LINEFINDER_LEFT_ARROW_BIT);

or even better a union-struct

**Pull request merged**